collectors/scm/github-graphql/src/main/java/com/capitalone/dashboard/collector/DefaultGitHubClient.java

```diff
@@ -1042,6 +1056,42 @@ private static DateTime getDate(DateTime dateInstance, int offsetDays, int offse
         return dateInstance.minusDays(offsetDays).minusMinutes(offsetMinutes);
     }
 
+
+    private JSONObject getDataFromRestCallPost(String url, GitHubRepo repo, String password, String personalAccessToken, JSONObject query) throws MalformedURLException, HygieiaException {
+        ResponseEntity<String> response = makeRestCallPost(url, repo.getUserId(), password, personalAccessToken, query);
+        JSONObject data = (JSONObject) parseAsObject(response).get("data");
+        JSONArray errors = getArray(parseAsObject(response), "errors");
+
+        if (!CollectionUtils.isEmpty(errors)) {
+            JSONObject error = (JSONObject) errors.get(0);
+            if (error.get("type").equals("NOT_FOUND")) {
+                RedirectedStatus redirectedStatus = checkForRedirectedRepo(repo);
+
+                if (redirectedStatus.isRedirected()) {
+                    String redirectedUrl = redirectedStatus.getRedirectedUrl();
+                    LOG.debug("Repo was redirected from: " + repo.getRepoUrl() + " to " + redirectedUrl);
+                    repo.setRepoUrl(redirectedUrl);
+                    GitHubParsed gitHubParsed = new GitHubParsed(redirectedUrl);
+
+                    JSONParser parser = new JSONParser();
+                    try {
+                        JSONObject variableJSON = (JSONObject) parser.parse((String) query.get("variables"));
+                        variableJSON.put("name", gitHubParsed.getRepoName());
+                        variableJSON.put("owner", gitHubParsed.getOrgName());
+                        query.put("variables", variableJSON.toString());
+                    } catch (ParseException e) {
+                        LOG.error("Could not parse JSON String", e);
+                    }
+                    return getDataFromRestCallPost(url, repo, password, personalAccessToken, query);
+                }
+            }
+
```

Comment:
The code handles one level of redirection. Not sure we need to handle 2 or more levels of redirections?

Reply:
As far as I can tell from trying it out with test repos, all of the repos will get redirected to the end repo, so there should only be one level.

For example if I renamed a repo from org/repo1 -> org/repo2 -> org/repo3, the org/repo1 will redirect directly to org/repo3 in the `html_url` section of the response body.

Reply:
@andrewalvintran does  org/repo2  also point to  org/repo3?

Reply:
Yup, they all redirect to the same repo, which is org/repo3